core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/AbstractRdsConfigProvider.java

```diff
@@ -42,66 +41,97 @@
     @Inject
     private RedbeamsDbServerConfigurer dbServerConfigurer;
 
-    @Inject
-    private DatabaseCommon dbCommon;
-
     @Inject
     private RedbeamsClientService redbeamsClientService;
 
+    @Inject
+    private SecretService secretService;
+
+    /**
+     * Creates a map of database information for this provider's database, suitable for inclusion
+     * in a salt pillar. The map usually contains only one key, that being the return value of
+     * {@link #getPillarKey()}. The configuration information is ultimately used to create the
+     * database.
+     *
+     * @param  stack   stack for cluster
+     * @param  cluster cluster to associate database with
+     * @return         salt pillar configuration information for this provider's database
+     */
     public Map<String, Object> createServicePillarConfigMapIfNeeded(Stack stack, Cluster cluster) {
         if (isRdsConfigNeeded(cluster.getBlueprint())) {
             Set<RDSConfig> rdsConfigs = createPostgresRdsConfigIfNeeded(stack, cluster);
             RDSConfig rdsConfig = rdsConfigs.stream().filter(rdsConfig1 -> rdsConfig1.getType().equalsIgnoreCase(getRdsType().name())).findFirst().get();
             if (rdsConfig.getStatus() == ResourceStatus.DEFAULT && rdsConfig.getDatabaseEngine() != DatabaseVendor.EMBEDDED) {
                 Map<String, Object> postgres = new HashMap<>();
-                String db = getDb();
                 if (dbServerConfigurer.isRemoteDatabaseNeeded(cluster)) {
-                    JdbcConnectionUrlFields parsedUrl = dbCommon.parseJdbcConnectionUrl(rdsConfig.getConnectionURL());
-                    postgres.put("remote_db_url", parsedUrl.getHost());
-                    postgres.put("remote_db_port", parsedUrl.getPort());
-                    postgres.put("remote_admin", rdsConfig.getConnectionUserName());
-                    postgres.put("remote_admin_pw", rdsConfig.getConnectionPassword());
+                    DatabaseServerV4Response dbServerResponse = dbServerConfigurer.getDatabaseServer(cluster.getDatabaseServerCrn());
```

Comment:
You're right, there is a prior call in the control flow. This call is in order to get the administrative username and password of the server, because that information is not saved in the RDSConfig object that is created from the call in `RedbeamsDbServerConfigurer`.

I'll see if there is some refactoring that can be done so that this second call can be eliminated.

Reply:
Well, I did a bunch of refactoring, but I don't think it will help. I believe that, after the first call site, the data is only preserved in the core database, and then retrieved from there at the second call site. (I cannot tell which one comes first, but it does seem they are very far apart.) So, eliminating the second call requires adding the server information to RDSConfig. I would rather have that done as a separate task to improve performance, than try to do it along with all the other changes here.

Reply:
I though it would work the same as it worked for AWS postgres and every information is present in the rdsConfig.
What's in the RdsConfig itself or what's the difference?

Reply:
The RDSConfig contains the connection information for the database, like the Hive database. This now properly uses / stores a specific user for the database, and not the administrative username / password. Therefore, the administrative username and password have to be retrieved from the lookup, in redbeams, of the database server itself.

You are correct that this change is not specific to Azure. I changed what is saved in RDSConfig, in general, so that it is now the database-specific user, and not the administrative user.